lms/resources/_js_config.py

```diff
@@ -140,6 +143,31 @@ def _add_canvas_submission_params(self, **kwargs):
         # Add the given document_url or canvas_file_id.
         self.config["submissionParams"].update(kwargs)
 
+    def maybe_enable_grading(self):
```

Comment:
This still kills me, but I can't think of anything better. `try_to_enable_grading()`? At least we commit to trying?

Reply:
It's not "try to enable grading if we can", though, it's "enable grading if appropriate".

A name like `enable_grading_if_not_canvas()` might work but the _if_ here is if the user is an instructor, if the assignment is gradeable, and if the LMS is not Canvas. So it would end up having to be `enable_grading_if_appropriate()` or something.

Or you split it into two functions:

```python
if context.js_config.should_enable_grading():
    context.js_config.enable_grading()
```

But I don't really want to do that.

I actually don't mind the "maybe" naming style, I think it may be the best approach.

Reply:
To be clear, I mean kills me in that I think it's really funny. I don't object to it at all.